planner/optimize.go

```diff
@@ -427,9 +427,11 @@ func OptimizeExecStmt(ctx context.Context, sctx sessionctx.Context,
 }
 
 func buildLogicalPlan(ctx context.Context, sctx sessionctx.Context, node ast.Node, builder *core.PlanBuilder) (core.Plan, error) {
-	sctx.GetSessionVars().PlanID = 0
-	sctx.GetSessionVars().PlanColumnID = 0
-	sctx.GetSessionVars().MapHashCode2UniqueID4ExtendedCol = nil
+	if !sctx.GetSessionVars().StmtCtx.InHandleForeignKeyTrigger {
```

Comment:
Why we need this?

Reply:
Without this, it will reset the plan ID, and then the foreign key cascade plan maybe have duplicate plan ID, and the duplicate plan ID will cause collected executor runtime stats error.

Reply:
With out this, here is an example, As you can see, there are 2 `Delete_2`, and `Point_Get_1` has same plan ID with `Batch_Point_Get_1`, and the `execution info` are same for the executor has same plan ID.

```sql
test> explain analyze delete from t1 where id = 1;
+-------------------------+---------+---------+------+---------------+---------------------------------------------------------------------------------------------------+------------------------------------------+-----------+------+
| id                      | estRows | actRows | task | access object | execution info                                                                                    | operator info                            | memory    | disk |
+-------------------------+---------+---------+------+---------------+---------------------------------------------------------------------------------------------------+------------------------------------------+-----------+------+
| Delete_2                | N/A     | 0       | root |               | time:232.5µs, loops:2                                                                             | N/A                                      | 148 Bytes | N/A  |
| ├─Point_Get_1           | 1.00    | 2       | root | table:t1      | time:153.2µs, loops:4, BatchGet:{num_rpc:1, total_time:12.8µs},Get:{num_rpc:1, total_time:58.7µs} | handle:1                                 | N/A       | N/A  |
| └─Foreign_Key_Cascade_3 | 0.00    | 0       | root | table:t2      | total:111µs, foreign_keys:1                                                                       | foreign_key:fk, on_delete:CASCADE        | N/A       | N/A  |
|   └─Delete_2            | N/A     | 0       | root |               | time:232.5µs, loops:2                                                                             | N/A                                      | 148 Bytes | N/A  |
|     └─Batch_Point_Get_1 | 1.00    | 2       | root | table:t2      | time:153.2µs, loops:4, Get:{num_rpc:1, total_time:58.7µs},BatchGet:{num_rpc:1, total_time:12.8µs} | handle:[1], keep order:false, desc:false | N/A       | N/A  |
+-------------------------+---------+---------+------+---------------+---------------------------------------------------------------------------------------------------+------------------------------------------+-----------+------+
```